sklearn/utils/testing.py

```diff
@@ -372,14 +375,52 @@ def __exit__(self, *exc_info):
     assert_greater = _assert_greater
 
 
+def _sparse_dense_allclose(val1, val2, rtol=1e-7, atol=0):
```

Comment:
I was also thinking of adding this to scipy itself and removing off sklearn... Does that sound good? But not sure they will receive this well at scipy :/